src/app/multi-editor/multi-editor.component.ts

```diff
@@ -147,15 +154,19 @@ export class MultiEditorComponent implements OnInit {
       error => { this.errorMessage   = error; this.changeDetectorRef.markForCheck(); });
   }
 
-  filterNewRecord(record: object): object {
+  filterRecord(record: object , objectToAssign: object): object {
     let _record = Object.assign({}, record);
     if (this.filterExpression) {
-      return this.jsonUtilsService.filterObject(_record, [this.filterExpression]);
+      let filtered = this.jsonUtilsService.filterObject(_record, [this.filterExpression]);
+      objectToAssign = filtered;
+      return filtered;
     }
+    objectToAssign = record;
     return record;
   }
 
   private queryCollection(query, collection) {
+    this.previewMode = false;
```

Comment:
do this before where you call this function, also for readability reasons, put callee function right after the caller function.